js-setup/js/component-initializer.js

```diff
@@ -0,0 +1,153 @@
+import ads from '../../ads';
+import tracking from '../../tracking';
+import date from '../../date';
+import header from '../../header';
+import promoMessages from '../../promo-messages';
+import cookieMessage from '../../cookie-message';
+import welcomeMessage from '../../welcome-message';
+import messagePrompts from '../../message-prompts';
+import footer from '../../footer';
+import myft from '../../myft';
+
+export const presets = {
+	discrete: {
+		header: true,
+		footer: true,
+		date: true,
+		welcomeMessage: true
+	},
+	complete: {
+		header: true,
+		footer: true,
+		date: true,
+		cookieMessage: true,
+		welcomeMessage: true,
+		myft: true,
+		messagePrompts: true,
+		promoMessages: false,
+		ads: true
+	}
+};
+
+export class ComponentInitializer {
+	constructor () {
+		this.initializedFeatures = {};
+		this.configuration = {};
+		this.loader = new Loader();
+	}
+
+	isInitialized (feature) {
+		return !!this.initializedFeatures[feature];
+	}
+
+	configure (options = {}) {
+		// NOTE: just store configuration for now, need to wait for polyfill to load before assigning
+		console.log('n-ui.configure is deprecated - pass in your config object as a first argument to bootstrap instead')
+		this.configuration = options;
+	}
+
+	bootstrap (config, cb) {
+		// backwards compatible with previous signature of bootstrap(cb);
+		if (!cb && typeof config === 'function') {
+			cb = config;
+			config = null;
+		}
+
+		cb = cb || (() => null);
+		config = config || this.configuration;
+
+		// belt and braces backwards compatibility for old api, which expected a flat config object
+		if (!config.features) {
+			config.features = Object.assign({}, config);
+		}
+
+		config.features = Object.assign({}, presets[config.preset], config.features);
+
+		return this.loader.bootstrap(config, ({ flags, mainCss, appInfo }) => { // eslint-disable-line
+
+			if (!this.isInitialized('tracking')) {
+				// FT and next tracking
+				tracking.init(flags, appInfo);
+				this.initializedFeatures.tracking = true;
+			}
+
+			if (config.features.myft && !this.isInitialized('myftclient')) {
+				const clientOpts = [];
+
+				if (flags.get('follow')) {
+					clientOpts.push({relationship: 'followed', type: 'concept'});
+				}
+
+				if (flags.get('saveForLater')) {
+					clientOpts.push({relationship: 'saved', type: 'content'});
+				}
+				myft.client.init(clientOpts);
+
+				this.initializedFeatures.myftClient = true
+			}
+
+			if (config.features.header && !this.isInitialized('header')) {
+				header.init(flags);
+				this.initializedFeatures.header = true;
+			}
+
+			if (config.features.footer && !this.isInitialized('footer')) {
+				footer.init(flags);
+				this.initializedFeatures.footer = true
+			}
+
+			if (config.features.date && !this.isInitialized('date')) {
+				date.init();
+				this.initializedFeatures.date = true
+			}
+
+			mainCss
+				.then(() => {
+					if (config.features.cookieMessage && !this.isInitialized('cookieMessage')) {
```

Comment:
the method call doesn't seem to add much convenience, maybe it could take a second arg to set? 

```js
this.initialized('cookieMessage') // false
this.initialized('cookieMessage', true) // sets to true
```